### DISCUSSION TITLE
4.6.1 has issue to parse password from stdout

### DISCUSSION THREAD

**Discussion by nephaste** [userdocs/qbittorrent-nox-static#126] [General]
Hi,

I am using your releases for Qnap NAS package since years now, and porpose it as QPKG to community

latest version has breaking change with password parsing in stdout (when scripted)

maybe you are already aware

https://github.com/qbittorrent/qBittorrent/issues/19984

**Comment by saltydk** [comment]
If you want to script it just set the password by generating the login fields in the config.

**Comment by nephaste** [comment]
thanks will check for this solution, may be source will be fixed in 4.6.2

**Comment by saltydk** [comment]
Some proof of concept ways of generating the password hash:
```bash
#!/bin/bash

# Check if a password is provided
if [ "$#" -ne 1 ]; then
    echo "Usage: $0 <password>"
    exit 1
fi

# Parameters
PLAIN_PASSWD="$1"
ITERATIONS=100000
SALT_SIZE=16
KEY_LEN=64  # Length of the hash output in bytes (512 bits for SHA-512)

# Generate a random salt
SALT=$(openssl rand -hex $SALT_SIZE)

# Use OpenSSL KDF command for PBKDF2 hashing
HASH=$(openssl kdf -keylen $KEY_LEN -kdfopt digest:SHA512 -kdfopt pass:"$PLAIN_PASSWD" \
        -kdfopt salt:"$SALT" -kdfopt iter:$ITERATIONS PBKDF2 | tr -d '\n')

# Encode the salt and hash in Base64
SALT_BASE64=$(echo -n "$SALT" | xxd -r -p | base64 | tr -d '\n')
HASH_BASE64=$(echo -n "$HASH" | xxd -r -p | base64 | tr -d '\n')

# Output the formatted result
echo "@ByteArray(${SALT_BASE64}:${HASH_BASE64})"
```
Or with a light binary such as my placeholder repo here https://github.com/saltydk/qbt_pw_gen

**Reply by snowdream** [reply]
Do you have a try?( the bash script)

I have a try, but i find the password is wrong.

I do not know why?

**Reply by saltydk** [reply]
> Do you have a try?( the bash script)
> 
> I have a try, but i find the password is wrong.
> 
> I do not know why?

It works fine as it uses the same parameters as the qbittorrent source code. I suspect it is special characters and shell escapes that is your issue but given that you provide zero information it is hard to deal in specifics.

**Comment by userdocs** [comment]
If they patch it I can apply the patch and build a revision or they might drop a new version at the same time specifically for this fix.

Otherwise this is something that can be scripted around for deployment and is not something I would try to address here.

Backporting some patches is something I consider acceptable but I'm not deploying containers or anything here, just the binary.

@saltydk has provided a sensible approach to handling this deployment issue, thanks for that.

**Comment by userdocs** [comment]
I created a revision with these backports https://github.com/qbittorrent/qBittorrent/pull/19979 so the current build of 4.6.1 is fixed.
I guess 4.6.2 will drop at some point soon anyway though.
